n3fit/src/n3fit/model_trainer.py

```diff
@@ -999,6 +991,15 @@ def hyperparametrizable(self, params):
                     for penalty in self.hyper_penalties
                 }
 
+                hyper_loss_per_replica = experimental_loss + sum(penalties.values())
+                fold_loss = self._hyper_loss.reduce_over_replicas(hyper_loss_per_replica)
+                passed = fold_loss < self.hyper_threshold
```

Comment:
I guess these should averaged over replicas and folds, to make sure that the threshold does not depend on the choice (of replicas/folds)

Reply:
It's already reduced over replicas, in whatever way specified, so it shouldn't depend on the number of replicas. Regarding folds, I think there is also a normalization with respect to the number of datasets, no? So it should be fine?
You can also do it after all folds are done as you suggest, but then you won't be able to exit a trial early if the first fold was disastrous.

Reply:
Then you should not get to 1000...